nflow-engine/src/main/java/io/nflow/engine/workflow/curated/CronWorkflow.java

```diff
@@ -0,0 +1,132 @@
+    schedule(start), doWork(normal), handleFailure(normal), failed(manual);
+
+    private WorkflowStateType type;
+
+    State(WorkflowStateType type) {
+      this.type = type;
+    }
+
+    @Override
+    public WorkflowStateType getType() {
+      return type;
+    }
+
+    @Override
+    public String getDescription() {
+      return name();
+    }
+  }
+
+  /**
+   * Extend cron workflow definition with customized workflow settings.
+   * Extending class must implement the 'doWork` step method.
+   *
+   * @param type The type of the workflow.
+   * @param settings The workflow settings.
+   */
+  protected CronWorkflow(String type, WorkflowSettings settings) {
+    super(type, schedule, handleFailure, settings);
+    permit(schedule, doWork);
+    permit(doWork, schedule);
+    permit(handleFailure, schedule, failed);
+  }
+
+  /**
+   * Extend cron workflow definition.
+   * Extending class must implement the 'doWork` step method.
+   *
+   * @param type The type of the workflow.
+   */
+  protected CronWorkflow(String type) {
+    this(type, new Builder().setHistoryDeletableAfter(days(45)).build());
+  }
+
+  /**
+   * Calculates next activation time based on cron state variable. Override for custom scheduling.
+   *
+   * @param execution The workflow execution context.
+   * @param cron The cron schedule.
+   * @return The next activation time.
+   */
+  protected DateTime getNextActivationTime(@SuppressWarnings("unused") StateExecution execution, String cron) {
+    return new DateTime(new CronSequenceGenerator(cron).next(now().toDate()));
+  }
+
+  /**
+   * Logs an error and continues. Override for custom error handling.
+   *
+   * @param execution The workflow execution context.
+   * @return True if the failure was handled successfully. If false is returned the cron workflow stops waiting for manual actions.
+   */
+  protected boolean handleFailureImpl(StateExecution execution) {
+    logger.error("Cron workflow {} / {} work failed", getType(), execution.getWorkflowInstanceId());
+    return true;
+  }
+
+  /**
+   * Determines the next execution time for the doWork step.
+   *
+   * @param execution The workflow execution context.
+   * @param cron The cron state variable.
+   * @return Action to run the doWork step at scheduled time.
+   */
+  public NextAction schedule(StateExecution execution, @StateVar(value = VAR_SCHEDULE, readOnly = true) String cron) {
+    return moveToStateAfter(doWork, getNextActivationTime(execution, cron), "Scheduled");
+  }
+
+  /**
+   * Tries to handle failures by calling {@link #handleFailureImpl}
+   *
+   * @param execution The workflow execution context.
+   * @return Either moves back to schedule state or if failure handling failed waits for manual actions in failed state.
+   */
+  public NextAction handleFailure(StateExecution execution) {
+    if (handleFailureImpl(execution)) {
+      return moveToState(schedule, "Failure handled successfully");
```

Comment:
We don't know if the failure was handled successfully, it is up to handleFailureImpl to decide how to continue regardless of the failure handling status.
```suggestion
      return moveToState(schedule, "Failure handled, rescheduling");
```